### DISCUSSION TITLE
Docker Compose Storage Question

### DISCUSSION THREAD

**Discussion by geodesic2910** [oae/kaizoku#87] [General]
I have tried using the docker config and used bind mounts for the config/library/etc.
But the library doesn't appear to work properly. In every instance where it's downloading the cbz files it's storing them in an Overlay2.
I'm still relatively new to docker so I may be misunderstanding something but I thought the library volume is supposed to be where it stores the downloaded files?

**Comment by oae** [comment]
After you mount your manga path like  `/hostmachinepathtomanga:/data`, you should select the same directory on UI when you first set it up. It will ask you to select a library path. You should write `/data`. Mangas will be downloaded to this path. Also, make sure that you have the correct permissions for that mount. You can pass `PUID`, and `PGUID` env variables to achieve that.

**Reply by eswiig** [reply]
~~how can I change the path? I typed in data/test as a test for the library and now that I've tried it out I want to set it as /data~~

edit: disregard, figured it out, deleted the volumes

**Comment by geodesic2910** [comment]
Could you possibly help me figure out what I am doing wrong because I have even made the user owner of the mounting point /Media as a test and I am feeling like a bit of an idiot at the moment.

`version: '3'

volumes:
  db:
  redis:

services:
  app:
    container_name: kaizoku
    image: ghcr.io/oae/kaizoku:latest
    environment:
      - DATABASE_URL=postgresql://kaizoku:kaizoku@db:5432/kaizoku
      - KAIZOKU_PORT=3000
      - REDIS_HOST=redis
      - REDIS_PORT=6379
      - PUID=1000
      - PGID=1000
      - TZ=Europe/Istanbul
    volumes:
      - /Media:/data
      - /mnt/Storage1/AppData/kaizoku/config:/config
      - /mnt/Storage1/AppData/kaizoku/logs:/logs
    depends_on:
      db:
        condition: service_healthy
    ports:
      - '3000:3000'
  redis:
    image: redis:7-alpine
    volumes:
      - redis:/data
  db:
    image: postgres:alpine
    restart: unless-stopped
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -U kaizoku']
      interval: 5s
      timeout: 5s
      retries: 5
    environment:
      - POSTGRES_USER=kaizoku
      - POSTGRES_DB=kaizoku
      - POSTGRES_PASSWORD=kaizoku
    volumes:
      - db:/var/lib/postgresql/data
madmin@nas-store:~$ sudo docker compose up
[+] Running 36/31
 ✔ db 8 layers [⣿⣿⣿⣿⣿⣿⣿⣿]      0B/0B      Pulled                                    15.3s 
 ✔ app 20 layers [⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿]      0B/0B      Pulled                      28.9s 
 ✔ redis 5 layers [⣿⣿⣿⣿⣿]      0B/0B      Pulled                                    13.6s 
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
                                                                                          
[+] Running 6/5
 ✔ Network madmin_default    Created                                                 0.1s 
 ✔ Volume "madmin_redis"     Created                                                 0.0s 
 ✔ Volume "madmin_db"        Created                                                 0.0s 
 ✔ Container madmin-db-1     Created                                                 0.3s 
 ✔ Container madmin-redis-1  Created                                                 0.3s 
 ✔ Container kaizoku         Created                                                 0.0s 
Attaching to kaizoku, madmin-db-1, madmin-redis-1
madmin-redis-1  | 1:C 04 Apr 2023 19:31:22.849 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
madmin-redis-1  | 1:C 04 Apr 2023 19:31:22.849 # Redis version=7.0.10, bits=64, commit=00000000, modified=0, pid=1, just started
madmin-redis-1  | 1:C 04 Apr 2023 19:31:22.849 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
madmin-redis-1  | 1:M 04 Apr 2023 19:31:22.849 * monotonic clock: POSIX clock_gettime
madmin-redis-1  | 1:M 04 Apr 2023 19:31:22.850 * Running mode=standalone, port=6379.
madmin-redis-1  | 1:M 04 Apr 2023 19:31:22.850 # Server initialized
madmin-redis-1  | 1:M 04 Apr 2023 19:31:22.850 # WARNING Memory overcommit must be enabled! Without it, a background save or replication may fail under low memory condition. Being disabled, it can can also cause failures without low memory condition, see https://github.com/jemalloc/jemalloc/issues/1328. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
madmin-redis-1  | 1:M 04 Apr 2023 19:31:22.850 * Ready to accept connections
madmin-db-1     | The files belonging to this database system will be owned by user "postgres".
madmin-db-1     | This user must also own the server process.
madmin-db-1     | 
madmin-db-1     | The database cluster will be initialized with locale "en_US.utf8".
madmin-db-1     | The default database encoding has accordingly been set to "UTF8".
madmin-db-1     | The default text search configuration will be set to "english".
madmin-db-1     | 
madmin-db-1     | Data page checksums are disabled.
madmin-db-1     | 
madmin-db-1     | fixing permissions on existing directory /var/lib/postgresql/data ... ok
madmin-db-1     | creating subdirectories ... ok
madmin-db-1     | selecting dynamic shared memory implementation ... posix
madmin-db-1     | selecting default max_connections ... 100
madmin-db-1     | selecting default shared_buffers ... 128MB
madmin-db-1     | selecting default time zone ... UTC
madmin-db-1     | creating configuration files ... ok
madmin-db-1     | running bootstrap script ... ok
madmin-db-1     | sh: locale: not found
madmin-db-1     | 2023-04-04 19:31:23.291 UTC [30] WARNING:  no usable system locales were found
madmin-db-1     | performing post-bootstrap initialization ... ok
madmin-db-1     | syncing data to disk ... ok
madmin-db-1     | 
madmin-db-1     | 
madmin-db-1     | Success. You can now start the database server using:
madmin-db-1     | 
madmin-db-1     |     pg_ctl -D /var/lib/postgresql/data -l logfile start
madmin-db-1     | 
madmin-db-1     | initdb: warning: enabling "trust" authentication for local connections
madmin-db-1     | initdb: hint: You can change this by editing pg_hba.conf or using the option -A, or --auth-local and --auth-host, the next time you run initdb.
madmin-db-1     | waiting for server to start....2023-04-04 19:31:23.784 UTC [36] LOG:  starting PostgreSQL 15.2 on x86_64-pc-linux-musl, compiled by gcc (Alpine 12.2.1_git20220924-r4) 12.2.1 20220924, 64-bit
madmin-db-1     | 2023-04-04 19:31:23.786 UTC [36] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
madmin-db-1     | 2023-04-04 19:31:23.789 UTC [39] LOG:  database system was shut down at 2023-04-04 19:31:23 UTC
madmin-db-1     | 2023-04-04 19:31:23.791 UTC [36] LOG:  database system is ready to accept connections
madmin-db-1     |  done
madmin-db-1     | server started
madmin-db-1     | CREATE DATABASE
madmin-db-1     | 
madmin-db-1     | 
madmin-db-1     | /usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
madmin-db-1     | 
madmin-db-1     | waiting for server to shut down....2023-04-04 19:31:23.930 UTC [36] LOG:  received fast shutdown request
madmin-db-1     | 2023-04-04 19:31:23.932 UTC [36] LOG:  aborting any active transactions
madmin-db-1     | 2023-04-04 19:31:23.933 UTC [36] LOG:  background worker "logical replication launcher" (PID 42) exited with exit code 1
madmin-db-1     | 2023-04-04 19:31:23.934 UTC [37] LOG:  shutting down
madmin-db-1     | 2023-04-04 19:31:23.934 UTC [37] LOG:  checkpoint starting: shutdown immediate
madmin-db-1     | 2023-04-04 19:31:23.970 UTC [37] LOG:  checkpoint complete: wrote 918 buffers (5.6%); 0 WAL file(s) added, 0 removed, 0 recycled; write=0.033 s, sync=0.002 s, total=0.037 s; sync files=250, longest=0.001 s, average=0.001 s; distance=4222 kB, estimate=4222 kB
madmin-db-1     | 2023-04-04 19:31:23.975 UTC [36] LOG:  database system is shut down
madmin-db-1     |  done
madmin-db-1     | server stopped
madmin-db-1     | 
madmin-db-1     | PostgreSQL init process complete; ready for start up.
madmin-db-1     | 
madmin-db-1     | 2023-04-04 19:31:24.049 UTC [1] LOG:  starting PostgreSQL 15.2 on x86_64-pc-linux-musl, compiled by gcc (Alpine 12.2.1_git20220924-r4) 12.2.1 20220924, 64-bit
madmin-db-1     | 2023-04-04 19:31:24.049 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
madmin-db-1     | 2023-04-04 19:31:24.049 UTC [1] LOG:  listening on IPv6 address "::", port 5432
madmin-db-1     | 2023-04-04 19:31:24.051 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
madmin-db-1     | 2023-04-04 19:31:24.055 UTC [52] LOG:  database system was shut down at 2023-04-04 19:31:23 UTC
madmin-db-1     | 2023-04-04 19:31:24.058 UTC [1] LOG:  database system is ready to accept connections
kaizoku         | [migrations] started
kaizoku         | [migrations] no migrations found
kaizoku         | cont-init: info: running /etc/cont-init.d/40-config
kaizoku         | 
kaizoku         | -------------------------------------
kaizoku         |           _         ()
kaizoku         |          | |  ___   _    __
kaizoku         |          | | / __| | |  /  \
kaizoku         |          | | \__ \ | | | () |
kaizoku         |          |_| |___/ |_|  \__/
kaizoku         | 
kaizoku         | 
kaizoku         | Brought to you by linuxserver.io
kaizoku         | -------------------------------------
kaizoku         | 
kaizoku         | To support LSIO projects visit:
kaizoku         | https://www.linuxserver.io/donate/
kaizoku         | -------------------------------------
kaizoku         | GID/UID
kaizoku         | -------------------------------------
kaizoku         | 
kaizoku         | User uid:    1000
kaizoku         | User gid:    1000
kaizoku         | -------------------------------------
kaizoku         | 
kaizoku         | Prisma schema loaded from prisma/schema.prisma
kaizoku         | Datasource "db": PostgreSQL database "kaizoku", schema "public" at "db:5432"
madmin-db-1     | 2023-04-04 19:31:30.071 UTC [63] LOG:  could not receive data from client: Connection reset by peer
kaizoku         | 
kaizoku         | 5 migrations found in prisma/migrations
kaizoku         | 
kaizoku         | Applying migration `20221015234159_init`
kaizoku         | Applying migration `20221027101522_fix_intervals`
kaizoku         | Applying migration `20221103223653_add_settings_schema`
kaizoku         | Applying migration `20221124200902_add_kavita`
kaizoku         | Applying migration `20230125230322_add_out_of_sync_chapter_table`
kaizoku         | 
kaizoku         | The following migrations have been applied:
kaizoku         | 
kaizoku         | migrations/
kaizoku         |   └─ 20221015234159_init/
kaizoku         |     └─ migration.sql
kaizoku         |   └─ 20221027101522_fix_intervals/
kaizoku         |     └─ migration.sql
kaizoku         |   └─ 20221103223653_add_settings_schema/
kaizoku         |     └─ migration.sql
kaizoku         |   └─ 20221124200902_add_kavita/
kaizoku         |     └─ migration.sql
kaizoku         |   └─ 20230125230322_add_out_of_sync_chapter_table/
kaizoku         |     └─ migration.sql
kaizoku         |       
kaizoku         | All migrations have been successfully applied.
kaizoku         | cont-init: info: /etc/cont-init.d/40-config exited 0
kaizoku         | [custom-init] No custom files found, skipping...
kaizoku         | services-up: info: copying legacy longrun kaizoku (no readiness notification)
kaizoku         | [ls.io-init] done.
kaizoku         | ✓ set metadata.comic_info_xml to true
kaizoku         | ✓ set metadata.series_json to true
kaizoku         | ✓ set formats.use to cbz
kaizoku         | ✓ set downloader.download_cover to true
kaizoku         | ✓ set downloader.redownload_existing to true
kaizoku         | ✓ set downloader.chapter_name_template to [{padded-index}] {chapter}
kaizoku         | ✓ set downloader.create_manga_dir to true
kaizoku         | ✓ set downloader.create_volume_dir to false
kaizoku         | ✓ set downloader.stop_on_error to true
kaizoku         | ✓ set downloader.path to .
kaizoku         | prisma:query SELECT "public"."Manga"."id", "public"."Manga"."createdAt", "public"."Manga"."title", "public"."Manga"."interval", "public"."Manga"."source", "public"."Manga"."libraryId", "public"."Manga"."metadataId" FROM "public"."Manga" WHERE 1=1 OFFSET $1
kaizoku         | [22:31:43.118] INFO (307): > Ready on http://localhost:3000 - env production`
